job/server/src/main/java/alluxio/job/plan/load/LoadDefinition.java

```diff
@@ -74,22 +74,37 @@ public LoadDefinition() {
     List<BlockWorkerInfo> workers = new ArrayList<>();
     for (BlockWorkerInfo worker : context.getFsContext().getCachedWorkers()) {
       if (jobWorkersByAddress.containsKey(worker.getNetAddress().getHost())) {
+        String workerHost = worker.getNetAddress().getHost().toUpperCase();
+        if ((!isEmptySet(config.getExcludeWorkerSet()))
+            && config.getExcludeWorkerSet().contains(workerHost)) {
+          continue;
+        }
         // If specified the locality id, the candidate worker must match one at least
         boolean match = false;
-        if (!isEmptySet(config.getLocalityIds())) {
-          if (worker.getNetAddress().getTieredIdentity().getTiers() != null) {
+        if (worker.getNetAddress().getTieredIdentity().getTiers() != null) {
+          if (!(isEmptySet(config.getLocalityIds())
+              && isEmptySet(config.getExcludeLocalityIds()))) {
+            boolean exclude = false;
             for (LocalityTier tier : worker.getNetAddress().getTieredIdentity().getTiers()) {
-              if (config.getLocalityIds().contains(tier.getValue().toUpperCase())) {
+              if (!isEmptySet(config.getExcludeLocalityIds())
```

Comment:
L86 `&& isEmptySet(config.getExcludeLocalityIds()` means `config.getExcludeLocalityIds()` is empty,
so this check must be false right?